addons/mail/static/src/model/model_listener.js

```diff
@@ -30,9 +30,21 @@ export class Listener {
         this.isPartOfUpdateCycle = isPartOfUpdateCycle;
         this.name = name;
         this.onChange = onChange;
-        // Set of last observed elements, for debugging purposes.
+        /**
+         * Set of last observed localIds.
+         * Useful to be able to remove this listener from them without iterating.
+         */
         this.lastObservedLocalIds = new Set();
-        this.lastObservedFields = new Set();
+        /**
+         * Map between last observed localIds and a Set of fields last observed on them.
+         * Useful to be able to remove this listener from them without iterating.
+         */
+        this.lastObservedFieldsByLocalId = new Map();
+        /**
+         * Set of last observed "all" by Model.
+         * Useful to be able to remove this listener from them without iterating.
+         */
+        this.lastObservedAllByModel = new Set();
```

Comment:
These comments lack context. What is "last"? What is the time window that is relevant for these dynamic attributes? "Removing listener without iterating" also lack context, so it's hard to understand why we need this.

A good comment should put much more emphasis on the "what". We don't really care on the "why", could be deduced by code, and a good "what" gives us ideas for "why". Both are lacking, especially the "what". So it's all obscure to me.

Reply:
Maybe we should avoid using "last" in these context, it's too imperative and also that's what makes it hard to understand.

Reply:
I updated the docstring with more details.
Those fields already existed, I'd rather not rename them here. (Except for the Model one but I'd rather follow the same logic as the others).

Reply:
Before these changes, it was only used to help debugging, so I didn't care too much on the naming at the time. But now that model manager make use of these registries, this becomes a significant code maintenance problem.